core/src/Kokkos_Macros.hpp

```diff
@@ -601,4 +601,102 @@ define KOKKOS_FORCEINLINE_FUNCTION inline
 #define KOKKOS_ENABLE_CUDA_LDG_INTRINSIC
 #endif
 
+//----------------------------------------------------------------------------
+// Thread sanitizer and intel inspector compiler macros
+
+#ifndef KOKKOS_THREAD_SANITIZER_IGNORE
+#if defined(__has_feature)
+#if __has_feature(thread_sanitizer)
+#define KOKKOS_THREAD_SANITIZER_IGNORE __attribute__((no_sanitize("thread")))
```

Comment:
lets make that a KOKKOS_IMPL macro. I don't think we want to encourage Kokkos users to litter this all over the place. In the end I believe if thread_sanitizer finds something in user code they probably violated our semantics andyway.

Reply:
sounds good